codetransformer/instructions.py

```diff
@@ -80,19 +76,26 @@ def __new__(mcls, name, bases, dict_, *, opcode=None):
             raise TypeError('Invalid opcode: {}'.format(opcode))
 
         opname_ = opname[opcode]
-        dict_['opcode'] = _immutableattr(opcode)
-        dict_['absjmp'] = _immutableattr(opcode in hasjabs)
-        dict_['reljmp'] = _immutableattr(opcode in hasjrel)
-        dict_['opname'] = _immutableattr(opname[opcode])
-        dict_['uses_name'] = _immutableattr(opname_ in _uses_name)
-        dict_['uses_varname'] = _immutableattr(opname_ in _uses_varname)
-        dict_['uses_free'] = _immutableattr(opname_ in _uses_free)
-        dict_['have_arg'] = _immutableattr(opcode >= HAVE_ARGUMENT)
+        dict_['opcode'] = immutableattr(opcode)
+        dict_['absjmp'] = immutableattr(opcode in hasjabs)
+        dict_['reljmp'] = immutableattr(opcode in hasjrel)
+        dict_['opname'] = immutableattr(opname[opcode])
+        dict_['uses_name'] = immutableattr(opname_ in _uses_name)
+        dict_['uses_varname'] = immutableattr(opname_ in _uses_varname)
+        dict_['uses_free'] = immutableattr(opname_ in _uses_free)
+        dict_['have_arg'] = immutableattr(opcode >= HAVE_ARGUMENT)
         cls = mcls._type_cache[opcode] = super().__new__(
             mcls, opname[opcode], bases, dict_,
         )
         return cls
 
+    def mcompile(self):
```

Comment:
I like that we could change to some other backend without worrying. the fact that the compiled matches are regular expressions is only an implementation detail